module/icp/core/kcf_mech_tabs.c

```diff
@@ -664,6 +664,9 @@ kcf_remove_mech_provider(char *mech_name, kcf_provider_desc_t *prov_desc)
 
 	mutex_exit(&mech_entry->me_mutex);
 
+	if (prov_mech == NULL)
+		return;
+
```

Comment:
I agree with @behlendorf about moving this to the default case. This defect came about because the original Illumos code simply does not have a default case.